### ISSUE TITLE
[Feature Request] #pragma once

### ISSUE THREAD

**Issue by jeffythedragonslayer** [cc65/cc65#1596]
#pragma once

**Comment by mrdudz** [comment]
say what

**Comment by jeffythedragonslayer** [comment]
https://docs.microsoft.com/en-us/cpp/preprocessor/once?view=msvc-170

**Comment by polluks** [comment]
Not M$ centric POV https://stackoverflow.com/questions/787533/is-pragma-once-a-safe-include-guard

**Comment by mrdudz** [comment]
Just for curiousity - what kind of project is it where this is relevant? I have a rather large C project that i work on, and compiling is pretty much instant, on my not so great PC :)

**Comment by jeffythedragonslayer** [comment]
I was going through the book "Making Games for the NES" and found a place where directives were being put in comments and parsed, so I thought pragma support for this platform could be improved.

**Comment by skeetor** [comment]
Would also be a nice enhancement for CA65. ;)

**Comment by under4mhz** [comment]
I'm trying to port some of my games from sdcc to cc65, and I use #pragma once exclusively in all my header files instead of include guards. I've found it to be neater and easier than include guards, being a single common keyword. Most modern compilers support it including gcc, clang, Visual Studio, z88dk and sdcc.

**Comment by groessler** [comment]
On 5/15/22 03:37, Under4Mhz wrote:
>
> I'm trying to port some of my games from sdcc to cc65, and I use 
> #pragma once exclusively in all my header files instead of include 
> guards. I've found it to be neater and easier than include guards, 
> being a single common keyword. Most modern compilers support it 
> including gcc, clang, Visual Studio, z88dk and sdcc.
>

My stance is "don't use '#pragma' if you can do without."

Call me old-fashioned... :-)

https://books.google.de/books?id=9f9uAQAAQBAJ&pg=PA31&lpg=PA31&dq=gcc+pragma+/usr/games&source=bl&ots=UK7hp2hLay&sig=ACfU3U3rWZeK6HNPwgFbcZ3x5LwMgVb5Mw&hl=en&sa=X&ved=2ahUKEwibmc3qguL3AhU2SfEDHRR_Bk0Q6AF6BAgeEAM#v=onepage&q=gcc%20pragma%20%2Fusr%2Fgames&f=false

**Comment by under4mhz** [comment]
Yes, you make a valid point, and not so long ago I would have agreed with you. But then I got old and I started doing what was easy instead.

**Comment by groessler** [comment]
Fair enough.

If patches are provided I guess it will be included.

**Comment by mrdudz** [comment]
> If patches are provided I guess it will be included.

please refer to the discussion in #2353
